### ISSUE TITLE
What's new with the new bot API?

### ISSUE THREAD

**Issue by santimacia** [yagop/telegram-bot#275]
How could change this project after the new telegram announcement:

https://telegram.org/blog/bot-revolution

**Comment by bb010g** [comment]
It's been fun, guys. :) Here's to less phone numbers!

**Comment by yagop** [comment]
It could continue like it is or change the way it receives and send the messages. Time to think guys, lets test the new API and decide!

**Comment by rockneurotiko** [comment]
Yeah! I'm reading it, let's see the pros/cons of change it.
The two main links are:
- [Introduction to bots](https://core.telegram.org/bots)
- [API](https://core.telegram.org/bots/api)

**Comment by payamohajeri** [comment]
This new API is really cool, its try to solve the privacy issues with bots in telegram group chats, also there is no need of phone numbers for every telegram bot. but I'm not okay with following part : 

**Bots have limited cloud storage — older messages may be removed by the server shortly after they have been processed.**

we now should store the message in our databases for getting histories.
I think this new API is going to be a BIG STEP for telegram guys.

also,I Just use " botfather " to create a bot, it's great. BTW I'm still searching for it's mother !! :)

**Comment by elboletaire** [comment]
I think that we could take some ideas, but both projects could stay alive in parallel.

For example, I loved the way the images bot answers you. It quotes you and shows the image after that. Thus it is clear who owns each message.

**Comment by payamohajeri** [comment]
I agree that we should keep alive this project and create a new project for the new API, cause they have different functionalities. what we have now, as a bot, behave like a normal user, it needs phone numbers and it can act like a user in receiving, reading and sending message. (as far as I know :) ).

but the newly implemented API, behaves so much limited, its capabilities depends on what you want to do, but generally I think It's more limited than the old one. since you can not access the all message ( which cause privacy issues :) ). but I really like the JSON responses in the new one.

I think that it's better that we ( I will help guys :D ) continue to develop this project, and defining a new project based on the new telegram API. both projects could have different pros/cons.

**Comment by luksi-reiku** [comment]
Same, I think we should keep two different projects, the new APi sounds very cool but as you said, its a bit limited.

**Comment by francesco-p** [comment]
The new API is very slow, and still limitating. I think that the major advantage respect this bot is the personalization of the user interface...and the official recognition of the status: bot. 

For me it's quite limitating, but this move will force a large part of dev (if not all) to create bot through the API without using a modified client. :(

**Comment by santimacia** [comment]
The new api is a bit limited now but the new features looks promising. 
If we can make the plugins compatible with the both projects, could be a great hit.

**Comment by luksi-reiku** [comment]
This may be useful: https://github.com/yukuku/telebot

**Comment by bb010g** [comment]
I like how we can deal with `/` commands better. Also, buttons. I think the main reason for not keeping history is so that if a bot is added to a group and everyone in the group leaves, they don't have to clean up after the bot. It doesn't "own" the media or text used in the cloud, the people do. When all the *people* in a group or conversation don't need media anymore, it can be deleted from the cloud. These bots we've made so far can't let go that easily.

**Comment by Benjarobbi** [comment]
Please dont use the new Bot API!! I like the bot with the normal account. Please make a second bot, if you really want the new Bot API :pray:

**Comment by hamednourhani** [comment]
we have some problem when requesting uploaded pic from telegram bot api . the bot server send two content type and akka-http throw exception ! 
`< HTTP/1.1 200 OK
* Server nginx/1.10.0 is not blacklisted
< Server: nginx/1.10.0
< Date: Thu, 24 Nov 2016 17:22:40 GMT
< Content-Type: image/jpeg
< Content-Length: 2441
< Connection: keep-alive
< Last-Modified: Wed, 23 Nov 2016 20:53:09 GMT
< ETag: "583601b5-989"
< Accept-Ranges: bytes
< Content-Disposition: attachment
< Content-Type: application/octet-stream`

**Comment by bb010g** [comment]
@hamednourhani This is not a Bot API support forum.

**Comment by bb010g** [comment]
This issue should probably be closed.

_Sent from my Motorola MotoG3 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by bb010g** [comment]
This issue should probably be closed now.
